pkg/virtctl/vm/vm_test.go

```diff
@@ -565,4 +595,79 @@ var _ = Describe("VirtualMachine", func() {
 			Entry("with dry-run arg", true),
 		)
 	})
+
+	Context("Expand command", func() {
+		BeforeEach(func() {
+			vm = kubecli.NewMinimalVM(vmName)
+		})
+
+		It("should succeed when called on vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewVirtctlCommand("expand", vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		})
+
+		DescribeTable("should succeed when called with ", func(formatName string) {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewVirtctlCommand("expand", outputFormat, formatName, vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		},
+			Entry("supported format json", "json"),
+			Entry("supported format yaml", "yaml"),
+		)
+
+		It("should fail when called on non existing vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(nonExistingVM).Return(nil, fmt.Errorf("\"%s\" not found", nonExistingVM))
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", nonExistingVM)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).Should(MatchError("error expanding VirtualMachine non-existing-vm: \"non-existing-vm\" not found"))
+		})
+
+		It("should fail when called with non supported output format", func() {
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", outputFormat, invalidFormat, vmName)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).Should(MatchError("error not supported output format defined: test-format"))
+		})
+	})
+
+	Context("Expand command with file input", func() {
+		BeforeEach(func() {
+			file, err = ioutil.TempFile("", "file-*")
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(os.WriteFile(file.Name(), []byte(vmSpec), 0777)).To(Succeed())
+			Expect(yaml.Unmarshal([]byte(vmSpec), vm)).To(Succeed())
+		})
+
+		It("should succeed when called with existing file input", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().ExpandSpec(testingNamespace).Return(expandSpecInterface).Times(1)
+			expandSpecInterface.EXPECT().ForVirtualMachine(vm).Return(vm, nil).Times(1)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, file.Name())
+			err := cmd()
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should fail when called with invalid vm defined in file input", func() {
```

Comment:
This case tests for seemingly valid YAML that can be unmarshalled but fails on the API. What about a test case for invalid YAML that cannot be unmarshalled?

Reply:
What about this test case? If too much effort you could also do it in a follow up.